### ISSUE TITLE
[Bug Report] generate config `build.lib.formats: ["umd"]`  broken vite `import.meta.glob` function

### ISSUE THREAD

**Issue by Rhilip** [aklinker1/vite-plugin-web-extension#20]
Vite provide a [Glob Import](https://vitejs.dev/guide/features.html#glob-import) feature , however when use this feature function `import.meta.glob` in script, the build will fail and with this output:

```
Invalid value "umd" for option "output.format" - UMD and IIFE output formats are not supported for code-splitting builds.
error during build:
Error: Invalid value "umd" for option "output.format" - UMD and IIFE output formats are not supported for code-splitting builds.
    at error (D:\path-to-project\node_modules\.pnpm\rollup@2.68.0\node_modules\rollup\dist\shared\rollup.js:159:30)
    at validateOptionsForMultiChunkOutput (D:\path-to-project\node_modules\.pnpm\rollup@2.68.0\node_modules\rollup\dist\shared\rollup.js:16134:16)
    at Bundle.generate (D:\path-to-project\node_modules\.pnpm\rollup@2.68.0\node_modules\rollup\dist\shared\rollup.js:15968:17)
    at processTicksAndRejections (node:internal/process/task_queues:96:5)
    at handleGenerateWrite (D:\path-to-project\node_modules\.pnpm\rollup@2.68.0\node_modules\rollup\dist\shared\rollup.js:23612:23)
    at doBuild (D:\path-to-project\node_modules\.pnpm\vite@2.8.5\node_modules\vite\dist\node\chunks\dep-ff3ab196.js:39134:26)
    at Object.build (D:\path-to-project\node_modules\.pnpm\vite@2.8.5\node_modules\vite\dist\node\chunks\dep-ff3ab196.js:38959:16)
    at buildScript (D:\path-to-project\node_modules\.pnpm\vite-plugin-web-extension@1.1.2_vite@2.8.5\node_modules\vite-plugin-web-extension\dist\index.js:90:3)
    at Object.closeBundle (D:\path-to-project\node_modules\.pnpm\vite-plugin-web-extension@1.1.2_vite@2.8.5\node_modules\vite-plugin-web-extension\dist\index.js:524:11)
    at async Promise.all (index 0)
Waiting for the debugger to disconnect...
 ELIFECYCLE  Command failed with exit code 1.
```

It's ok when use function `import.meta.globEager` since this function will not splitting code.

----------------

## Steps to reproduce

1. clone the last bugreport project https://github.com/Rhilip/bugreport-vite-plugin-web-extension-watchbuild 
2. create two or more files like `src/background/module/1.ts` and `src/background/module/2.ts` , each may have simple code like `console.log("hello world");`
3. in `src/background/index.ts` , add line `console.log(import.meta.glob("./module/*.ts"));`
4. run command `pnpm run build` and see error code, no `dist/background` folder will be built.

## Try to fix

- I debug and find this line https://github.com/aklinker1/vite-plugin-web-extension/blob/5faeb5af478d1468e0abf5a516d32edfc69c2a8c/lib/src/build-script.ts#L56  When I change it from `["umd"]` to `["es"]`, it worked as except, and `1.js`, `2.js`, `index.js` in output dir `dist/background`

--------------------

- I also find a related vite issue https://github.com/vitejs/vite/issues/2982 , it pointed out that we can add `build.rollupOptions.output.inlineDynamicImports: true` to avoid this problem, however it still not work if we change project `vite.config.ts` file, since build-script always use a empty rollupOptions. https://github.com/aklinker1/vite-plugin-web-extension/blob/5faeb5af478d1468e0abf5a516d32edfc69c2a8c/lib/src/build-script.ts#L45

**Comment by aklinker1** [comment]
Alright, I've implmented `libModeViteConfig` and released it in [`v1.3.0`](https://github.com/aklinker1/vite-plugin-web-extension/releases/tag/v1.3.0). For an example of configuring the lib mode builds for your case, see the tests:

https://github.com/aklinker1/vite-plugin-web-extension/blob/19a2c4a9a569286df10f8465a1128871209a1790/tests/e2e.test.ts#L325-L333

**Comment by aklinker1** [comment]
@Rhilip if that doesn't work for you, feel free to reopen with more details.
